server/sonar-web/src/main/js/apps/component-measures/details/actions.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { getMeasuresAndMeta } from '../../../api/measures';
+import { enhanceWithLeak } from '../utils';
+
+/*
+ * Actions
+ */
+
+export const REQUEST_MEASURE = 'details/REQUEST_MEASURE';
+export const RECEIVE_MEASURE = 'details/RECEIVE_MEASURE';
+
+
+/*
+ * Action Creators
+ */
+
+function requestMeasure (metric) {
+  return { type: REQUEST_MEASURE, metric };
+}
+
+function receiveMeasure (measure, secondaryMeasure, periods) {
+  return { type: RECEIVE_MEASURE, measure, secondaryMeasure, periods };
+}
+
+
+/*
+ * Workflow
+ */
+
+export function fetchMeasure (metricKey) {
+  return (dispatch, getState) => {
+    const { component, metrics } = getState().app;
+    const metricsToRequest = [metricKey];
+
+    if (metricKey === 'ncloc') {
+      metricsToRequest.push('ncloc_language_distribution');
+    }
+    if (metricKey === 'function_complexity') {
+      metricsToRequest.push('function_complexity_distribution');
+    }
+    if (metricKey === 'file_complexity') {
+      metricsToRequest.push('file_complexity_distribution');
+    }
+
+    const metric = metrics.find(metric => metric.key === metricKey);
+    dispatch(requestMeasure(metric));
+
+    getMeasuresAndMeta(
+        component.key,
+        metricsToRequest,
+        { additionalFields: 'periods' }
+    ).then(r => {
+      const measures = enhanceWithLeak(r.component.measures);
+      const measure = measures.find(measure => measure.metric === metricKey);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) "measure" hides or potentially hides a variable declared in an outer scope at line 72. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=javascript%3AVariableShadowing)